quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -400,14 +417,30 @@ public void validateExpectedValues(ConfigValue configValue, BiConsumer<ConfigVal
         String value = configValue.getValue();
 
         boolean multiValued = getOption().getType() == java.util.List.class;
+        StringBuilder result = new StringBuilder();
 
         String[] values = multiValued ? value.split(",") : new String[] { value };
         for (String v : values) {
             if (multiValued && !v.trim().equals(v)) {
-                throw new PropertyException("Invalid value for multivalued option " + getOptionAndSourceMessage(configValue)
+                if (!result.isEmpty()) {
+                    result.append(". ");
+                }
+                result.append("Invalid value for multivalued option " + getOptionAndSourceMessage(configValue)
                         + ": list value '" + v + "' should not have leading nor trailing whitespace");
+                continue;
             }
-            singleValidator.accept(configValue, v);
+            try {
+                singleValidator.accept(configValue, v);
+            } catch (PropertyException e) {
+                if (!result.isEmpty()) {
+                    result.append(". ");
```

Comment:
I like the validation errors accumulating. Very nice stuff. 

What I'd like more is how these errors will be displayed for users.

Instead of having them on a single line delimited by dot, I'd rather have them displayed as a list where each error will be on the new line (if there is more).

Sth like (example):
```
There are multiple validation errors:
* Invalid value for option '--users': invalid. Expected values are: skip, realm_file, same_file, different_files
* Property '--users' can be used only when exporting to a directory, or value set to 'same_file' when exporting to a file.
* Something else...
```

Reply:
> Instead of having them on a single line delimited by dot, I'd rather have them displayed as a list where each error will be on the new line (if there is more).

Sure, separate lines with a header is good.

And it's probably good to note that this is per option. We are not yet aggregating across options.

Reply:
> And it's probably good to note that this is per option. We are not yet aggregating across options.

Yes, I was thinking about it as well, but... In that case, we should edit the validator messages so as not to duplicate the property name again. However, if there's only one error, there will not be any header, and the message will not contain the property name. 

So it should be handled somehow - but yes, it's possible, probably not a big deal.

Reply:
Switched to newline delimited. The header would be nice, but may not be worth it at this point. Alternatively how the exceptions are collected / emitted could change - a validator could just as easily be a List<String> supplier which would make it easy for the top level validate method to emit a message with a header. I would leave that to a follow-up.